bratshelper/tests.go

```diff
@@ -196,14 +196,33 @@ func StagingWithADepThatIsNotTheLatest(depName string, copyBrats func(string) *c
 func StagingWithCustomBuildpackWithCredentialsInDependencies(depRegexp string, copyBrats func(string) *cutlass.App) {
 	Describe("staging with custom buildpack that uses credentials in manifest dependency uris", func() {
 		var (
-			buildpackFile, bpName, stack string
-			app                          *cutlass.App
+			buildpackFile, bpName, stack, username, password string
+			app                                              *cutlass.App
 		)
 		JustBeforeEach(func() {
 			file, err := ModifyBuildpackManifest(buildpackFile, func(m *Manifest) {
 				for _, d := range m.Dependencies {
 					uri, err := url.Parse(d.URI)
-					uri.User = url.UserPassword("login", "password")
+					if proxyHost, ok := os.LookupEnv("PROXY_HOST"); ok {
+						uri.Host = proxyHost
+					}
+					if proxyPort, ok := os.LookupEnv("PROXY_PORT"); ok {
+						uri.Host += ":" + proxyPort
+					}
+					if proxyScheme, ok := os.LookupEnv("PROXY_SCHEME"); ok {
+						uri.Scheme = proxyScheme
+					}
+					if proxyUsername, ok := os.LookupEnv("PROXY_USERNAME"); ok {
+						username = proxyUsername
+					} else {
+						username = "login"
+					}
+					if proxyPassword, ok := os.LookupEnv("PROXY_PASSWORD"); ok {
+						password = proxyPassword
+					} else {
+						password = "password"
+					}
+					uri.User = url.UserPassword(username, password)
```

Comment:
The rest of the tests in this file are using the string literals `"login"` and `"password"` to check that they are redacted from the output. This change as written will cause the tests to not actually test anything when `PROXY_USERNAME` and/or `PROXY_PASSWORD` are set.

Reply:
Good point, thanks. We were preparing another PR for those other tests because the regexp used there is also a problem for us. We will merge both fixes on this PR to fix this problem as well. Probably tomorrow since it is already late today.

Reply:
Problem fixed. Will open another PR for the regexp.